web-common/src/features/dashboards/state-managers/cascading-explore-state-merge.ts

```diff
@@ -0,0 +1,67 @@
+import type { MetricsExplorerEntity } from "@rilldata/web-common/features/dashboards/stores/metrics-explorer-entity";
+
+const ShallowMergeOneLevelDeepKeys = new Set<keyof MetricsExplorerEntity>([
+  "selectedTimeRange",
+  "selectedComparisonTimeRange",
+  "tdd",
+  "pivot",
+]);
```

Comment:
Now that I see this again, aren't these keys really at the _top-level_, not one-level-deep? So, would `TopLevelKeysToBeShallowMerged` be accurate?